### PR TITLE
#790: Create RandomText

### PR DESCRIPTION
**Pull request by proshin-roman**
Pull request for #790.

I've created a new class RandomText that generates a random text ("captain obvious" :D) allowing to specify length and characters of the resulting text. As well I've added a text for that class.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#830](https://github.com/yegor256/cactoos/issues/830) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#830](https://github.com/yegor256/cactoos/issues/830) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/C63314D6Z/1271718901); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/text/RandomText.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+     * @param len Length of generated text.
+     */
+    public RandomText(final List<Character> chrs, final int len) {
+        this.characters = chrs;
+        this.length = len;
+        this.random = new SecureRandom();
+    }
+
+    @Override
+    public String asString() throws IOException {
```
Comment (Vatavuk) [inline src/main/java/org/cactoos/text/RandomText.java]: @proshin-roman you don't need to throw exception here.

File: src/main/java/org/cactoos/text/RandomText.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            ),
+            len
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param chrs List of characters allowed for generating.
+     */
+    public RandomText(final List<Character> chrs) {
```
Comment (Vatavuk) [inline src/main/java/org/cactoos/text/RandomText.java]: @proshin-roman Add one more ctor `RandomText(final Character... chrs)` to simplify usage.

File: src/main/java/org/cactoos/text/RandomText.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            new SecureRandom().nextInt(RandomText.MAX_RANDOM_LENGTH - 1) + 1
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param chrs List of characters allowed for generating.
+     * @param len Length of generated text.
+     */
+    public RandomText(final List<Character> chrs, final int len) {
```
Comment (Vatavuk) [inline src/main/java/org/cactoos/text/RandomText.java]: @proshin-roman Add one more ctor `RandomText(final int len, final Character... chrs)` to simplify usage.
Also expand the main ctor to `RandomText(final List<Character> chrs, final int len, final Random random)` if the user wants to provide `Random` with a seed.

**Comment by Vatavuk** [comment]
@llorllale good to merge

**Comment by llorllale** [comment]
@proshin-roman our [principles](http://www.elegantobjects.org/) disallow "code in ctors". Please add a puzzle for internal refactor of `RandomText` so that calls to `SecureRandom` are deferred until `asString()` is invoked.

**Comment by proshin-roman** [comment]
@llorllale @Vatavuk I've decided to not create a puzzle but fix the issue myself - please check the code.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#830](https://github.com/yegor256/cactoos/issues/830) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @vatavuk[/z](https://www.0crat.com/u/vatavuk)

**Pull request merged**